encoder/vaapiencoder_h264.cpp

```diff
@@ -930,6 +930,15 @@ Encode_Status VaapiEncoderH264::reorder(const SurfacePtr& surface, uint64_t time
     bool isIdr = (m_frameIndex == 0 ||m_frameIndex >= keyFramePeriod() || forceKeyFrame);
 
     if (isIdr) {
+        // If the last frame before IDR is B frame, set it to P frame.
```

Comment:
As I verified, it will discard the last few B frames if last GOP has no enough frames. So the core dumped error won’t happen in such case. Maybe driver handles it already.

Best Regards,
Xiaoxia Zhong

From: Xu Guangxin [mailto:notifications@github.com]
Sent: Thursday, February 18, 2016 10:56 AM
To: 01org/libyami
Cc: Zhong, Xiaoxia
Subject: Re: [libyami] h264enc: Fix the encoding failure issue for IBBP mode. (#445)


In encoder/vaapiencoder_h264.cpp<https://github.com/01org/libyami/pull/445#discussion_r53266084>:

> @@ -930,6 +930,15 @@ Encode_Status VaapiEncoderH264::reorder(const SurfacePtr& surface, uint64_t time

>      bool isIdr = (m_frameIndex == 0 ||m_frameIndex >= keyFramePeriod() || forceKeyFrame);

>

>      if (isIdr) {

> +        // If the last frame before IDR is B frame, set it to P frame.

What if last gop do not have enough frame? Suppose you have IBBBBP pattern, but you only have two frame ...
Similar logical should apply to last gop too.

—
Reply to this email directly or view it on GitHub<https://github.com/01org/libyami/pull/445/files#r53266084>.